spark/src/main/java/com/netflix/iceberg/spark/source/IcebergSource.java

```diff
@@ -109,10 +113,18 @@ protected SparkSession lazySparkSession() {
     return lazySpark;
   }
 
-  protected Configuration lazyConf() {
+  protected Configuration lazyBaseConf() {
     if (lazyConf == null) {
       this.lazyConf = lazySparkSession().sparkContext().hadoopConfiguration();
     }
     return lazyConf;
   }
+
+  protected Configuration mergeIcebergHadoopConfs(Configuration baseConf, Map<String, String> options) {
```

Comment:
Is this method meant to be overridden in subclasses? Or would it be simpler to make it static private?

Also, would it be more simpler to not modify the pass in configuration [baseconf] and return a new Configuration object? It would make the usage simpler IMO. For instance, in line 60 we create a copy of baseconf before passing and in line 62, as we modify the passed in configuration, withTableConfs and conf end up being the same, which I'm not sure was intended or not

Reply:
No, this should not be overridden. Probably should be `private static` then.

Creating a new configuration is expensive. As long as the details are handled in a single class then it is fine to create just one copy and merge configs into it. It may be more clear that's the behavior if this didn't return a Configuration and returned void instead, but I think that's minor.